lib/services/consumed/autocomplete.js

```diff
@@ -0,0 +1,130 @@
+/* @flow */
+import { CompositeDisposable, Disposable } from "atom";
+
+import type { Store } from "../../store";
+import type WatchesStore from "../../store/watches";
+import type WatchStore from "../../store/watch";
+
+/**
+ * This acts as a global storage for the consumed service.
+ */
+export class AutocompleteWatchEditor {
+  disposables = new CompositeDisposable();
+  /** The `consumed autocompleteWatchEditor` */
+  addAutocompleteToEditor = (
+    editor: atom$TextEditor,
+    labels: Array<string>
+  ) => {
+    return;
+  };
+
+  /**
+   * This function is called on activation of autocomplete, or if autocomplete is
+   * already active, then it is called when hydrogen activates.
+   *
+   * @param {Store} store - The global Hydrogen store.
+   * @param {Function} watchEditor - The function provided by `autocomplete.watchEditor`.
+   * @returns {Disposable} This is for clean up when autocomplete or hydrogen deactivate.
+   */
+  consume(store: Store, watchEditor: atom$AutocompleteWatchEditor) {
+    this.addAutocompleteToEditor = watchEditor;
+
+    // Add autocomplete capabilities to already existing watches
+    for (let kernel of store.runningKernels) {
```

Comment:
It's just a talk of code style, but maybe `forEach` is preferred since we usually use that in most part of the other code base.

But if you prefer native `for` iterator for some reason, go with your style :wink:

Reply:
just because its nested I think it is more clear